Backpack/src/main/java/net/skyscanner/backpack/calendar/presenter/BpkCalendarController.kt

```diff
@@ -27,15 +27,32 @@ abstract class BpkCalendarController {
 
   abstract val locale: Locale
 
+  var selectionType: SelectionType = SelectionType.RANGE
```

Comment:
I think there's a potential bug here. We probably need to clear the selection if the selection type changes, otherwise it will keep a range selected even after the type was set to `SINGLE`. Maybe a better approach would be to make this immutable and a constructor argument and if users need a different selection they can instantiate and set a new controller (which will reset the selection)

Reply:
Not sure I am getting this.
`selectionType` gets initialised here as a `RANGE` as I won't need any change in already existing code. On normal usage, it will be initialised once only.
For the purpose of being of exposing a switcher in the sample calendars this is mutable.
If I make this immutable, no switcher would be possible. This would mean creating maybe 2 other `Story` with single day selection.
I am fine either ways. But as iOS has this sort of switcher I thought I would provide it.

Reply:
Or you mean re-instatiating the controller every time the selection is made?

Reply:
Yes I meant re-instantiating the controller to change the selection type and making it immutable, so something like:
```kotlin
view.controller = Controller(context) // Default is range
// latter on if selection type changes
view.controller = Controller(context, selectionType)
````
The problem right now is that if the controller is instantiated with `RANGE` and then latter one is changed to `SINGLE` a range will still be selected, which may not be a big problem but to be consistent we should reset the selected dates when the selection type changes.

Reply:
👍